mesonbuild/modules/cmake.py

```diff
@@ -48,6 +49,13 @@ class WriteBasicPackageVersionFile(T.TypedDict):
         name: str
         version: str
 
+    class ConfigurePackageConfigFile(T.TypedDict):
```

Comment:
`TypedDict` isn't in typing until python 3.8, so I think we still need to use `typing_extensions.TypedDict`

Reply:
Oh ok, I thought that the `if T.TYPE_CHECKING` was enough, as the tests are passing in the Ubuntu Bionic. Should I change that also in the `WriteBasicPackageVersionFile` class introduced in #9916?

Reply:
It will work if you run with mypy based on python 3.8 or later. I don’t think we’ve ever had a clear policy on whether we support all versions with mypy we do with in general or not.

Reply:
Done :)